shell/platform/darwin/common/framework/Source/FlutterStandardCodecHelper.c

```diff
@@ -0,0 +1,189 @@
+    case FlutterStandardFieldString:
+    case FlutterStandardFieldUInt8Data:
+    case FlutterStandardFieldInt32Data:
+    case FlutterStandardFieldInt64Data:
+    case FlutterStandardFieldFloat64Data:
+    case FlutterStandardFieldList:
+    case FlutterStandardFieldMap:
+    case FlutterStandardFieldFloat32Data:
+      return true;
+    default:
+      return false;
+  }
+}
+
+void FlutterStandardCodecHelperReadBytes(unsigned long* location,
+                                         unsigned long length,
+                                         void* destination,
+                                         CFDataRef data) {
+  CFRange range = CFRangeMake(*location, length);
+  CFDataGetBytes(data, range, destination);
+  *location += length;
+}
+
+uint8_t FlutterStandardCodecHelperReadByte(unsigned long* location,
+                                           CFDataRef data) {
+  uint8_t value;
+  FlutterStandardCodecHelperReadBytes(location, 1, &value, data);
+  return value;
+}
+
+uint32_t FlutterStandardCodecHelperReadSize(unsigned long* location,
+                                            CFDataRef data) {
+  uint8_t byte = FlutterStandardCodecHelperReadByte(location, data);
+  if (byte < 254) {
+    return (uint32_t)byte;
+  } else if (byte == 254) {
+    UInt16 value;
+    FlutterStandardCodecHelperReadBytes(location, 2, &value, data);
+    return value;
+  } else {
+    UInt32 value;
+    FlutterStandardCodecHelperReadBytes(location, 4, &value, data);
+    return value;
+  }
+}
+
+static CFDataRef ReadDataNoCopy(unsigned long* location,
+                                unsigned long length,
+                                CFDataRef data) {
+  CFDataRef result = CFDataCreateWithBytesNoCopy(
+      kCFAllocatorDefault, CFDataGetBytePtr(data) + *location, length,
+      kCFAllocatorNull);
+  *location += length;
+  return CFAutorelease(result);
+}
+
+CFStringRef FlutterStandardCodecHelperReadUTF8(unsigned long* location,
+                                               CFDataRef data) {
+  uint32_t size = FlutterStandardCodecHelperReadSize(location, data);
+  CFDataRef bytes = ReadDataNoCopy(location, size, data);
+  CFStringRef result = CFStringCreateFromExternalRepresentation(
+      kCFAllocatorDefault, bytes, kCFStringEncodingUTF8);
+  return CFAutorelease(result);
+}
+
+// Peeks ahead to see if we are reading a standard type.  If so, recurse
+// directly to FlutterStandardCodecHelperReadValueOfType, otherwise recurse to
+// objc.
+static inline CFTypeRef FastReadValue(
+    unsigned long* location,
+    CFDataRef data,
+    CFTypeRef (*ReadValue)(CFTypeRef),
+    CFTypeRef (*ReadTypedDataOfType)(FlutterStandardField, CFTypeRef),
+    CFTypeRef user_data) {
+  uint8_t type = PeekByte(*location, data);
+  if (IsStandardType(type)) {
+    *location += 1;
+    return FlutterStandardCodecHelperReadValueOfType(
+        location, data, type, ReadValue, ReadTypedDataOfType, user_data);
+  } else {
+    return ReadValue(user_data);
+  }
+}
```

Comment:
Can we have a unit test in the engine to test this? Basically making sure overriding Standard codec works? 
(We have a test in framework, but it would be nice to catch potential bugs in the engine)

Reply:
good call, done